### ISSUE TITLE
Android: java.lang.NullPointerException: Attempt to invoke virtual method 'org.tensorflow.Output org.tensorflow.Operation.output(int)' o

### ISSUE THREAD

**Issue by chengsam** [tensorflow/tensorflow#8233]
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
StackOverflow: [Why am I getting error: Initializing Tensorflow?](http://stackoverflow.com/questions/42386650/why-am-i-getting-error-initializing-tensorflow)

### Environment info
Operating System: Mac OS 10.11.6
Tensorflow version: v0.12.0

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
I have successfully built the Android demo from command line using bazel. After that, I replaced the inception model with my custom model. I have changed the parameters in `ClassifierActivity` with the following:
```
private static final int INPUT_SIZE = 299;
  private static final int IMAGE_MEAN = 128;
  private static final float IMAGE_STD = 128;
  private static final String INPUT_NAME = "Mul:0";
  private static final String OUTPUT_NAME = "final_result:0";
```
Then I build again with bazel, when I try to run the app, the app crashed. Logs are attached below.

### What other attempted solutions have you tried?
Using the same model with memory mapping works fine on iOS.

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

> FATAL EXCEPTION: main
>                                                                      Process: org.tensorflow.demo, PID: 23532
>                                                                      java.lang.RuntimeException: Error initializing TensorFlow!
>                                                                          at org.tensorflow.demo.ClassifierActivity.onPreviewSizeChosen(ClassifierActivity.java:131)
>                                                                          at org.tensorflow.demo.CameraActivity$1.onPreviewSizeChosen(CameraActivity.java:158)
>                                                                          at org.tensorflow.demo.CameraConnectionFragment.setUpCameraOutputs(CameraConnectionFragment.java:394)
>                                                                          at org.tensorflow.demo.CameraConnectionFragment.openCamera(CameraConnectionFragment.java:411)
>                                                                          at org.tensorflow.demo.CameraConnectionFragment.access$000(CameraConnectionFragment.java:63)
>                                                                          at org.tensorflow.demo.CameraConnectionFragment$1.onSurfaceTextureAvailable(CameraConnectionFragment.java:94)
>                                                                          at android.view.TextureView.getHardwareLayer(TextureView.java:370)
>                                                                          at android.view.View.updateDisplayListIfDirty(View.java:14115)
>                                                                          at android.view.View.getDisplayList(View.java:14161)
>                                                                          at android.view.View.draw(View.java:14928)
>                                                                          at android.view.ViewGroup.drawChild(ViewGroup.java:3405)
>                                                                          at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3199)
>                                                                          at android.view.View.updateDisplayListIfDirty(View.java:14133)
>                                                                          at android.view.View.getDisplayList(View.java:14161)
>                                                                          at android.view.View.draw(View.java:14928)
>                                                                          at android.view.ViewGroup.drawChild(ViewGroup.java:3405)
>                                                                          at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3199)
>                                                                          at android.view.View.draw(View.java:15207)
>                                                                          at android.widget.FrameLayout.draw(FrameLayout.java:592)
>                                                                          at android.view.View.updateDisplayListIfDirty(View.java:14138)
>                                                                          at android.view.View.getDisplayList(View.java:14161)
>                                                                          at android.view.View.draw(View.java:14928)
>                                                                          at android.view.ViewGroup.drawChild(ViewGroup.java:3405)
>                                                                          at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3199)
>                                                                          at android.view.View.updateDisplayListIfDirty(View.java:14133)
>                                                                          at android.view.View.getDisplayList(View.java:14161)
>                                                                          at android.view.View.draw(View.java:14928)
>                                                                          at android.view.ViewGroup.drawChild(ViewGroup.java:3405)
>                                                                          at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3199)
>                                                                          at android.view.View.updateDisplayListIfDirty(View.java:14133)
>                                                                          at android.view.View.getDisplayList(View.java:14161)
>                                                                          at android.view.View.draw(View.java:14928)
>                                                                          at android.view.ViewGroup.drawChild(ViewGroup.java:3405)
>                                                                          at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3199)
>                                                                          at android.view.View.draw(View.java:15207)
>                                                                          at android.widget.FrameLayout.draw(FrameLayout.java:592)
>                                                                          at com.android.internal.policy.impl.PhoneWindow$DecorView.draw(PhoneWindow.java:2599)
>                                                                          at android.view.View.updateDisplayListIfDirty(View.java:14138)
>                                                                          at android.view.View.getDisplayList(View.java:14161)
>                                                                          at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:268)
>                                                                          at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:274)
>                                                                          at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:313)
>                                                                          at android.view.ViewRootImpl.draw(ViewRootImpl.java:2503)
>                                                                          at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2341)
>                                                                          at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1972)
>                                                                          at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1058)
>                                                                          at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5796)
>                                                                          at android.view.Choreographer$CallbackRecord.run(Choreographer.java:767)
>                                                                          at android.view.Choreographer.doCallbacks(Choreographer.java:580)
>                                                                          at android.view.Choreographer.doFrame(Choreographer.java:550)
>                                                                          at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
>                                                                          at android.os.Handler.handleCallback(Handler.java:739)
>                                                                          at android.os.Handler.dispatchMessage(Handler.java:95)
>                                                                          at android.os.Looper.loop(Looper.java:211)
>                                                                          at android.app.ActivityThread.main(ActivityThread.java:5317)
>                                                                          at java.lang.reflect.Method.invoke(Native Method)
>                                                                          at java.lang.reflect.Method.invoke(Method.java:372)
>                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1016)
>                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
>                                                                       Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'org.tensorflow.Output org.tensorflow.Operation.output(int)' o

**Comment by andrewharp** [comment]
@chengsam This is the error typically seen when the output node is not found when the classifier tries to check the number of output classes.

Does this work if you remove the ":0"s from the end of the node names?

**Comment by chengsam** [comment]
@andrewharp It worked! But why the doc said the label names need the ":0"?

**Comment by andrewharp** [comment]
@chengsam It used to accept them before we switched to using the Java API internally. Looks like the comment just didn't get updated, so I'll make a fix.

**Comment by andrewharp** [comment]
To clarify, specifying ports on the node names passed to fillNode* and readNode* in TensorFlowInferenceInterface should still work.

The crash here was likely happening in TensorFlowImageClassifier where it tries to get the size of the output node prior to inference:
`final Operation operation = c.inferenceInterface.graph().operation(outputName);`

**Comment by MICROSOFT55555555hh** [comment]
I use keras change to tensorflow but when i use it on Android It says can't find the input layer but in keras my input layer is bidirectional_3 but I use keras.input_names result is sequential_3_input ,when I try both ,None of them how can I do??
